lib/Doctrine/DBAL/Connection.php

```diff
@@ -1332,6 +1291,18 @@ public function executeCacheQuery($sql, $params, $types, QueryCacheProfile $qcp)
 
         $stmt->setFetchMode($this->defaultFetchMode);
 
+        return $this->createV3CompatStatement($stmt);
+    }
+
+    /**
+     * @return ResultStatement&V3Result
+     */
+    private function createV3CompatStatement(ResultStatement $stmt)
```

Comment:
Please use "forward compatibility" instead of "v3". Also, the method doesn't unconditionally create a statement as the name implies. Would something like `ensureStatementForwardCompatibility()` make sense.

Reply:
Done